Composer/packages/ui-plugins/select-skill-dialog/src/SelectSkillDialogField.tsx

```diff
@@ -23,64 +21,45 @@ export const settingReferences = (skillName: string) => ({
 
 export const SelectSkillDialogField: React.FC<FieldProps> = (props) => {
   const { value, onChange } = props;
-  const { shellApi, skills = [] } = useShellApi();
-  const { addSkillDialog, displayManifestModal } = shellApi;
-  const [comboboxTitle, setComboboxTitle] = useState<string | null>(null);
-
-  const skillId = getSkillNameFromSetting(value?.skillEndpoint);
-  const { content, manifestUrl, name } = skills.find(({ id }) => id === skillId) || ({} as Skill);
 
-  const options: IComboBoxOption[] = skills.map(({ id, name }) => ({
-    key: name,
-    text: name,
-    data: settingReferences(id),
-    isSelected: id === skillId,
-  }));
+  const { shellApi, skills } = useShellApi();
+  const { displayManifestModal } = shellApi;
 
-  options.push(
-    {
-      key: 'separator',
-      itemType: SelectableOptionMenuItemType.Divider,
-      text: '',
-    },
-    { key: ADD_DIALOG, text: formatMessage('Add a new Skill Dialog') }
-  );
+  const skillNameIdentifier = getSkillNameFromSetting(value?.skillEndpoint);
+  const { manifest, name }: Skill = skills[skillNameIdentifier] || {};
 
-  if (comboboxTitle) {
-    options.push({ key: 'customTitle', text: comboboxTitle });
+  const options: IComboBoxOption[] = [];
+  for (const key in skills) {
+    const skill = skills[key];
+    const option = {
+      key: key,
+      text: skill.name,
+      data: settingReferences(key),
+      isSelected: key === skillNameIdentifier,
+    };
+    options.push(option);
   }
 
   const handleChange = (_, option: IComboBoxOption) => {
     if (option) {
-      if (option.key === ADD_DIALOG) {
-        setComboboxTitle(formatMessage('Add a new Skill Dialog'));
-        addSkillDialog().then((skill) => {
-          if (skill?.manifestUrl && skill?.name) {
-            onChange({ ...value, ...settingReferences(skill.name) });
-          }
-          setComboboxTitle(null);
-        });
-      } else {
-        onChange({ ...value, ...option.data });
-      }
+      onChange({ ...value, ...option.data });
     }
   };
 
   return (
     <React.Fragment>
       <ComboBoxField
```

Comment:
If we are going to remove the 'Add skill` option, we should use a Dropdown instead of a combo box.